File: web/src/api/kubernetes/serviceAccount.js
import service from '@/utils/request'

// 获取serviceAccount list
export const getServiceAccountList = (data) => {
  return service({
    url: '/serviceAccount/getServiceAccountList',
    method: 'post',
    data
  })
}

// 获取serviceAccount编排
export const getServiceAccountRaw = (data) => {
  return service({
    url: '/serviceAccount/getServiceAccountRaw',
    method: 'post',
    data
  })
}

// 获取serviceAccount in detail
export const getServiceAccountDetail = (data) => {
  return service({
    url: '/serviceAccount/getServiceAccountDetail',
    method: 'post',
    data
  })
}

// 删除serviceAccount
export const deleteServiceAccount = (data) => {
  return service({
    url: '/serviceAccount/deleteServiceAccount',
    method: 'post',
    data
  })
}

File: server/router/kubernetes/serviceAccount.go
package kubernetes

import (
	"github.com/gin-gonic/gin"
	v1 "github.com/pddzl/kubeops/server/api/v1"
	"github.com/pddzl/kubeops/server/middleware"
)

type ServiceAccountRouter struct{}

func (s *ServiceAccountRouter) InitServiceAccountRouter(Router *gin.RouterGroup) {
	serviceAccountRouter := Router.Group("serviceAccount").Use(middleware.OperationRecord())
	ServiceAccountRouterWithoutRecord := Router.Group("serviceAccount")
	serviceAccountApi := v1.ApiGroupApp.KubernetesApiGroup.ServiceAccountApi
	{
		serviceAccountRouter.POST("getServiceAccountDetail", serviceAccountApi.GetServiceAccountDetail) // 获取serviceAccount详情
		serviceAccountRouter.POST("getServiceAccountRaw", serviceAccountApi.GetServiceAccountRaw)       // 获取serviceAccount编排
		serviceAccountRouter.POST("deleteServiceAccount", serviceAccountApi.DeleteServiceAccount)       // 删除serviceAccount
	}
	{
		ServiceAccountRouterWithoutRecord.POST("getServiceAccountList", serviceAccountApi.GetServiceAccountList) // 获取所有serviceAccount
	}
}
